### ISSUE TITLE
Throws "Invalid hook call" error when used with styled-components

### ISSUE THREAD

**Issue by shubham-web** [geist-org/react#499]
# Bug report 🐞

## Version & Environment

  - Version of browser: Chrome 89.0.4389.114
  - Version of `geist-ui/react`: 2.1.4
  - Version of `next`: 10.0.9,
  - Version of `styled-components`: 5.2.1

## Expected Behaviour

I want to use styled-components for custom elements and for others [@geist-ui/react](https://www.npmjs.com/package/@geist-ui/react) but it's giving issues while setting up. 

According to [guide on server render](https://react.geist-ui.dev/en-us/guide/server-render) the _document.js and _app.js file has to be changed as shown. But when [styled-components](https://www.npmjs.com/package/styled-components) package is imported which also requires _document.js and _app.js files to be changed. The conflict happens. 

## Actual results (or Errors)

It's throwing an error saying "Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons: ..."

![image](https://user-images.githubusercontent.com/52674815/113188112-44ebce00-9277-11eb-9bdd-acc217ccb9d5.png)

### Here's my `_document.js`
```
import Document, { Html, Head, Main, NextScript } from "next/document";
import { ServerStyleSheet } from "styled-components";
import { CssBaseline } from "@geist-ui/react";

class MyDocument extends Document {
	static async getInitialProps(ctx) {
		const sheet = new ServerStyleSheet();
		const originalRenderPage = ctx.renderPage;
		const styles = CssBaseline.flush();

		try {
			ctx.renderPage = () =>
				originalRenderPage({
					enhanceApp: (App) => (props) => sheet.collectStyles(<App {...props} />),
				});

			const initialProps = await Document.getInitialProps(ctx);
			return {
				...initialProps,
				styles: (
					<>
						{initialProps.styles}
						{styles}
						{sheet.getStyleElement()}
					</>
				),
			};
		} finally {
			sheet.seal();
		}
	}
	render() {
		return (
			<Html lang="en">
				<Head>
					<meta charSet="utf-8" />
				</Head>
				<body>
					<Main />
					<NextScript />
				</body>
			</Html>
		);
	}
}

export default MyDocument;

```

### and `_app.js`
```
import Head from "next/head";
import { createGlobalStyle, ThemeProvider } from "styled-components";
import Layout from "./../components/Layout";
import { GeistProvider, CssBaseline } from "@geist-ui/react";

export default function App({ Component, pageProps }) {
	return (
		<>
			<Head>
				<meta name="viewport" content="width=device-width, initial-scale=1.0" />
			</Head>
			<GeistProvider>
				<CssBaseline />
				<GlobalStyle />
				<ThemeProvider theme={theme}>
					<Layout>
						<Component {...pageProps}  />
					</Layout>
				</ThemeProvider>
			</GeistProvider>
		</>
	);
}

const GlobalStyle = createGlobalStyle`
	body{
		margin: 0;
	}
`
```

### `.babelrc`
```
{
	"presets": ["next/babel"],
	"plugins": [["styled-components", { "ssr": true }]]
}

```

Upon running `npm ls react` the version it shows for geist is not the latest,
Not sure if it has any connection with this issue but sharing it here.

![image](https://user-images.githubusercontent.com/52674815/113194158-7025eb80-927e-11eb-8b19-8082a13f36c6.png)

**Comment by Deep-Codes** [comment]
@shubham-web  This works fine for me tho 

Can you reproduce and send a link?

Slightly Different Approach : [CodeSandbox](https://codesandbox.io/s/mystifying-mcnulty-hf1o1?file=/pages/_document.js)

**Comment by shubham-web** [comment]
Namaste Deepankar (@Deep-Codes), 
Here you go - [CodeSandBox](https://codesandbox.io/s/quirky-hooks-mqyqj?file=/package.json)
I'm using the latest version of packages (including react).

**Comment by Deep-Codes** [comment]
> Namaste Deepankar (@Deep-Codes),
> Here you go - [CodeSandBox](https://codesandbox.io/s/quirky-hooks-mqyqj?file=/package.json)
> I'm using the latest version of packages (including react).

<img width="1440" alt="Screenshot 2021-04-01 at 11 05 59 AM" src="https://user-images.githubusercontent.com/61158210/113248304-4bf7f800-92da-11eb-84b8-7b1a537ba14c.png">

<img width="793" alt="Screenshot 2021-04-01 at 11 07 23 AM" src="https://user-images.githubusercontent.com/61158210/113248366-721d9800-92da-11eb-8224-9d059c7da263.png">

Works on the CodeSandBox  and Locally Too 


🤔

**Comment by shubham-web** [comment]
Now it's working yeah. @geist-ui/react package is recently updated. 

Now when I run `npm ls react`
The version of react in geist's modules is the updated one -
![image](https://user-images.githubusercontent.com/52674815/113249333-22d86700-92dc-11eb-98e3-7802f60d056d.png)

Maybe had has something to do with it.

Btw Thanks for your help.

**Comment by flaming-codes** [comment]
Just as a note, I'm facing the same issue when trying to start a new Next.js-project with geist-ui. I'm using Node 12.20.0. Should I open a new issue for it?

<img width="372" alt="Bildschirmfoto 2021-04-09 um 23 32 53" src="https://user-images.githubusercontent.com/11635736/114242893-e962c400-998b-11eb-8b37-251f382c128d.png">

**Comment by Deep-Codes** [comment]
@flaming-codes i had done this before here's a [Code Sandbox Link](https://codesandbox.io/s/mystifying-mcnulty-hf1o1?file=/pages/_document.js) try it out if it doesn't work open a new issue but do provide a Reproduction to it.

**Comment by flaming-codes** [comment]
Not reproducible in CodeSandbox and on another machine of mine. Therefore it seems to be problem with my local installation / project setup.

**Comment by joshuanianji** [comment]
I'm facing the same issue with node `v15.12.0` when I'm starting a new Next.js project with geist-ui. @flaming-codes did you figure out any issues on your local installation that might have been the issue?

**Comment by flaming-codes** [comment]
@joshuanianji What I've done:

1.tried to install geist-ui with Node 12.20.x and NPM 7.x, which didn't work (also tried to use npm-shrinkwrap and manually override the peer-dep, with no success)
2. checked out my repo on another device with Node 10.x and NPM 6.x, install worked and I commited my changes
3. checked out the repo on my first device, everything worked out nicely

Note that I used `nvm` on both machines and also tried the install with Node 10.x on my first device initially, but that didn't work. Hope this helps.
